youtubeExtension/options.js

```diff
@@ -0,0 +1,86 @@
+function saveOptions() {
+
+  const checkboxIds = ['Profanity', 'Violence', 'Politics', 'Substances', 'Horrer'];
+  let options = {};
+
+  // loop through each checkbox id and save its state
+  checkboxIds.forEach(id => {
+    options[id] = document.getElementById(id).checked;
+  });
+
+  chrome.storage.sync.set(options);
+
+}
+
+function saveRange(){
+  const range = document.getElementById("my-range");
+  chrome.storage.sync.set({'my-range': slider.value});
+  console.log({'my-range': slider.value});
+}
+
+function restoreOptions() {
+
+  const checkboxIds = ['Profanity', 'Violence', 'Politics', 'Substances', 'Horrer'];
+  // retrieve the saved state of each checkbox and set it
+  chrome.storage.sync.get(checkboxIds, (items) => {
+    checkboxIds.forEach(id => {
+      document.getElementById(id).checked = items[id] || false;
+    });
+  });
+
+}
+
+function restoreRange(){
+  chrome.storage.sync.get('my-range', (items) => {
+    const slider = document.getElementById("my-range");
```

Comment:
Indentation is inconsistent here (and other places in this file). Among other things, this makes me think that a lot of the code in this file was generated by AI. That's not necessarily a problem, but I'm pointing it out just so you're aware of that if, for example, you're doing a take-home assignment when applying for a job.